copy/multiple.go

```diff
@@ -230,6 +230,16 @@ func (c *copier) copyMultipleImages(ctx context.Context) (copiedManifest []byte,
 			logrus.Debugf("Copying instance %s (%d/%d)", instance.sourceDigest, i+1, len(instanceCopyList))
 			c.Printf("Copying image %s (%d/%d)\n", instance.sourceDigest, i+1, len(instanceCopyList))
 			unparsedInstance := image.UnparsedInstance(c.rawSource, &instanceCopyList[i].sourceDigest)
+			singleImageOpts := copySingleImageOptions{requireCompressionFormatMatch: false}
+			if c.options.ForceCompressionFormat {
+				singleImageOpts.requireCompressionFormatMatch = true
+				// If CompressionFormat is nil set it to `Gzip` as default,
+				// since requireCompressionFormatMatch is pretty much `no-op`
+				// for `nil` CompressionFormat.
+				if c.options.DestinationCtx.CompressionFormat == nil {
```

Comment:
I think this case should fail instead; the user is asking for using _precisely_ $undefined. That just doesn’t make sense, and indicates the user made a mistake that the user would probably like to correct.

Reply:
When user specifies no `--compression` in CLI then all tools (`buildah` and `podman`) assumes that default is gzip. Consider a case where repo contains some other algo ( `zstd` ) but user wants to override it with a default `compression` so he does `buildah push --forcecompression test` but in this case he never selected any `--compression` but requested a `--forcecompression` but nothing will happen since same `zstd` will be reused again.

In such case `--forcecompression` did nothing. I am assuming if no compression is specified `--forcecompression` must select default compression or do you think, implementer ( buildah, podman ) should do it ?

Reply:
Ah nevermind if we error out user will be forced to provide a `--compression`